components/search/SearchCollection.vue

```diff
@@ -45,162 +45,130 @@
-  }
-
-  public created() {
-    this.initKeyboardEventHandler({
-      f: this.bindFilterEvents,
-    })
-  }
 
-  private bindFilterEvents(event) {
-    switch (event.key) {
-      case 'b':
-        this.updateListed(!this.vListed)
-        break
-      case 'n':
-        this.updateSortBy(this.sortOption?.[0] || '')
-        break
-      case 'o':
-        this.updateSortBy(this.sortOption?.[1] || '')
-        break
-      case 'a':
-        this.updateSortBy(this.sortOption?.[2] || '')
-        break
-      case 'e':
-        this.updateSortBy(this.sortOption?.[3] || '')
-        break
-    }
-  }
-
-  get vListed(): boolean {
-    return this.listed
-  }
-
-  set vListed(listed: boolean) {
-    this.updateListed(listed)
-  }
-
-  get vOwned(): boolean {
-    return this.owned
-  }
-
-  set vOwned(owned: boolean) {
-    this.updateOwned(owned)
-  }
-
-  get searchQuery(): string {
-    return this.search
-  }
+const emit = defineEmits([
+  'update:listed',
+  'update:owned',
+  'update:type',
+  'update:sortBy',
+  'update:search',
+])
+
+const preferencesStore = usePreferencesStore()
+const { replaceUrl } = useReplaceUrl()
+
+const route = useRoute()
+const router = useRouter()
+const { $consola } = useNuxtApp()
+
+onMounted(() => {
+  exist(route.query.search, updateSearch)
+  exist(route.query.type, updateType)
+  exist(route.query.sortBy, updateSortBy)
+  exist(route.query.listed, updateListed)
+  exist(route.query.owned, updateOwned)
+
+  useKeyboardEvents({
+    f: bindFilterEvents,
+  })
+})
 
-  set searchQuery(value: string) {
-    this.updateSearch(value)
+const bindFilterEvents = (event: KeyboardEvent) => {
+  switch (event.key) {
+    case 'b':
+      updateListed(!vListed.value)
+      break
+    case 'n':
+      updateSortBy(props.sortOption?.[0] || '')
+      break
+    case 'o':
+      updateSortBy(props.sortOption?.[1] || '')
+      break
+    case 'a':
+      updateSortBy(props.sortOption?.[2] || '')
+      break
+    case 'e':
+      updateSortBy(props.sortOption?.[3] || '')
+      break
   }
+}
 
-  get typeQuery(): string {
-    return this.type
-  }
+const vListed = computed({
+  get: () => props.listed,
+  set: (value: boolean) => {
+    updateListed(value)
+  },
+})
 
-  set typeQuery(value: string) {
-    this.updateType(value)
-  }
+const vOwned = computed({
+  get: () => props.owned,
+  set: (value: boolean) => {
+    updateOwned(value)
+  },
+})
 
-  get replaceBuyNowWithYolo(): boolean {
-    return this.preferencesStore.getReplaceBuyNowWithYolo
-  }
+const searchQuery = computed({
+  get: (): string => props.search,
+  set: (value: string) => {
+    updateSearch(value)
+  },
+})
 
-  @Emit('update:listed')
-  @Debounce(50)
-  updateListed(value: string | boolean): boolean {
-    const queryValue = String(value) === 'true'
-    this.replaceUrl(queryValue, 'listed')
-    return queryValue
-  }
+const replaceBuyNowWithYolo = preferencesStore.getReplaceBuyNowWithYolo
```

Comment:
should be computed
```suggestion
const replaceBuyNowWithYolo = computed(() => preferencesStore.getReplaceBuyNowWithYolo)
```

Reply:
Yes originally it was. But it is unnecessary since Pinia state is already reactive & it is already a getter in pinia.